### PR TITLE
fix: improve error handling on non-ok response for metric server

### LINKED ISSUE #2317: Metrics Scaler: Improve Error Handling on not-ok response
### What's even going on?
The Metrics API Scaler is pretty quit on non-200 API responses. [(See this link for code)](https://github.com/kedacore/keda/blob/cbe929bcff6ee6d036283ad13423d6da083cf7e8/pkg/scalers/metrics_api_scaler.go#L214-L217)

At the moment we are using this for around 2000 scale targets and whenever something on API side goes wrong its a real hassle to even find out what the query was, when this is your response:
```log
2021-11-23T11:59:33.926Z    ERROR    metrics_api_scaler    Error when checking metric value: api returned 403    {"error": "api returned 403"} 
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).isScaledObjectActive
    /workspace/pkg/scaling/scale_handler.go:228
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
    /workspace/pkg/scaling/scale_handler.go:211
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
    /workspace/pkg/scaling/scale_handler.go:145
```

### Suggestion
Integrate the requested API endpoint and an optional response body into the scale error to get decent info on why the query didn't go through.

### Possible new log
```log
2021-11-23T11:59:33.926Z    ERROR    metrics_api_scaler    Error when checking metric value for /api/v1/files/on/server: api returned 403    {"error": "403 forbidden - cannot change into directory"} 
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).isScaledObjectActive
    /workspace/pkg/scaling/scale_handler.go:228
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
    /workspace/pkg/scaling/scale_handler.go:211
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
    /workspace/pkg/scaling/scale_handler.go:145
```

### Use-Case

* Improve debugging of big scaled systems
* No need to have access to logs of metric endpoint to debug

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Friedrich42**
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #2317
Added url to error string

### CODE REVIEW

**Comment by Friedrich42** [comment]
Fixes https://github.com/kedacore/keda/issues/2317

**Review by JorTurFer — COMMENTED** [review]
LGTM!
Thanks for this contribution! ❤️ 
Could you update the changelog too please?

File: pkg/scalers/metrics_api_scaler.go [hunk 1]
```diff
@@ -210,13 +210,13 @@ func (s *metricsAPIScaler) getMetricValue(ctx context.Context) (*resource.Quanti
 	if err != nil {
 		return nil, err
 	}
+	defer r.Body.Close()
 
 	if r.StatusCode != http.StatusOK {
-		msg := fmt.Sprintf("api returned %d", r.StatusCode)
+		msg := fmt.Sprintf("%s: api returned %d", r.Request.URL.String(), r.StatusCode)
```
Comment (zroubalik) [inline pkg/scalers/metrics_api_scaler.go:216]: Looking good, the only concern is, that we might be exposing confidental info. Might be worth to mask the url?
https://github.com/kedacore/keda/blob/877d0f6425452c4e2bc9bca8f21a7c48f386a26c/pkg/util/normalize_string.go#L43

Comment (Friedrich42) [reply pkg/scalers/metrics_api_scaler.go:216]: Might be a good idea, but how do I determine which info is confidential?

Comment (zroubalik) [reply pkg/scalers/metrics_api_scaler.go:216]: we can use the referenced function (or modify it a little bit) to mask domain and subdomain and keep the path.

Comment (Friedrich42) [reply pkg/scalers/metrics_api_scaler.go:216]: If we need only path, we can just go with r.Request.URL.Path instead of full URL

Comment (zroubalik) [reply pkg/scalers/metrics_api_scaler.go:216]: yeah, that's even better

**Review by zroubalik — COMMENTED** [review]
@Friedrich42 Please update changelog as well and would be great if you can add a simple unit test for this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,7 @@
 - **Datadog Scaler:** Validate query to contain `{` to prevent panic on invalid query ([#2625](https://github.com/kedacore/keda/issues/2625))
 - **Kafka Scaler** Make "disable" a valid value for tls auth parameter ([#2608](https://github.com/kedacore/keda/issues/2608))
 - **RabbitMQ Scaler:** Include `vhost` for RabbitMQ when retrieving queue info with `useRegex` ([#2498](https://github.com/kedacore/keda/issues/2498))
+- **Metric API Scaler** Improve Error Handling on not-ok response ([#2317](https://github.com/kedacore/keda/issues/2317))
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply CHANGELOG.md:42]: Please move this up, so the list is sorted alphabetically.

**Review by zroubalik — COMMENTED** [review]
Looking good, thanks for the test!
Just a nit in the changelog.

**Review by zroubalik — APPROVED** [review]
LGTM, great!

**Review by JorTurFer — APPROVED** [review]
LGTM!
Thanks for the improvement ❤️

**Pull request merged**